Importing my modules:

In [3]:
import requests
import pandas as pd

Pulling in my data from an API:

In [4]:
r = requests.get('http://universities.hipolabs.com/search?country=United+States')
data = r.json()
df = pd.DataFrame(data=data)
df.head()


domains        country alpha_two_code  \
0    [marywood.edu]  United States             US   
1  [lindenwood.edu]  United States             US   
2    [sullivan.edu]  United States             US   
3        [fscj.edu]  United States             US   
4      [xavier.edu]  United States             US   

                      web_pages state-province  \
0     [http://www.marywood.edu]           None   
1  [http://www.lindenwood.edu/]           None   
2       [https://sullivan.edu/]           None   
3       [https://www.fscj.edu/]           None   
4     [https://www.xavier.edu/]           None   

                                    name  
0                    Marywood University  
1                  Lindenwood University  
2                    Sullivan University  
3  Florida State College at Jacksonville  
4                      Xavier University

My first descriptive statement, which shows the number of rows and columns that exist in my dataframe:

In [12]:
df.shape

(4424, 6)

My second descriptive statement, which returns useful information such as count and unique:

In [6]:
df.describe()

domains        country alpha_two_code  \
count        4424           4424           4424   
unique       2193              1              1   
top     [ccc.edu]  United States             US   
freq           14           4424           4424   

                             web_pages state-province  \
count                             4424            116   
unique                            2209             20   
top     [http://www.coastalpines.edu/]   Pennsylvania   
freq                                 4             56   

                               name  
count                          4424  
unique                         2202  
top     Middlesex Community College  
freq                              4

Another descriptive statment that shows that only 116 values in the state-province column are not null:

In [13]:
df.count()

domains           4424
country           4424
alpha_two_code    4424
web_pages         4424
state-province     116
name              4424
dtype: int64

Using a query to list all rows that have Pennsylvania listed in the state-province column:

In [24]:
df[df['state-province'] == 'Pennsylvania']

domains        country alpha_two_code  \
14           [upmc.edu, upmc.com]  United States             US   
197                  [bloomu.edu]  United States             US   
243                     [cup.edu]  United States             US   
283                 [clarion.edu]  United States             US   
376                     [esu.edu]  United States             US   
379                [edinboro.edu]  United States             US   
463               [haverford.edu]  United States             US   
492                     [iup.edu]  United States             US   
530                [kutztown.edu]  United States             US   
577               [mansfield.edu]  United States             US   
613            [millersville.edu]  United States             US   
725                     [pct.edu]  United States             US   
726                     [psu.edu]  United States             US   
727                  [sl.psu.edu]  United States             US   
735               [pittstate.edu]  United States             US   
823                    [ship.edu]  United States             US   
830                     [sru.edu]  United States             US   
1074  [upenn.edu, seas.upenn.edu]  United States             US   
1076                   [pitt.edu]  United States             US   
1077               [upj.pitt.edu]  United States             US   
1078               [upb.pitt.edu]  United States             US   
1192                  [wcupa.edu]  United States             US   
1245                    [ycp.edu]  United States             US   
1977          [pennhighlands.edu]  United States             US   
1978                    [pit.edu]  United States             US   
1979                    [pti.edu]  United States             US   
1982               [upt.pitt.edu]  United States             US   
2187              [pointpark.edu]  United States             US   
2226         [upmc.edu, upmc.com]  United States             US   
2409                 [bloomu.edu]  United States             US   
2455                    [cup.edu]  United States             US   
2495                [clarion.edu]  United States             US   
2588                    [esu.edu]  United States             US   
2591               [edinboro.edu]  United States             US   
2675              [haverford.edu]  United States             US   
2704                    [iup.edu]  United States             US   
2742               [kutztown.edu]  United States             US   
2789              [mansfield.edu]  United States             US   
2825           [millersville.edu]  United States             US   
2937                    [pct.edu]  United States             US   
2938                    [psu.edu]  United States             US   
2939                 [sl.psu.edu]  United States             US   
2947              [pittstate.edu]  United States             US   
3035                   [ship.edu]  United States             US   
3042                    [sru.edu]  United States             US   
3286  [upenn.edu, seas.upenn.edu]  United States             US   
3288                   [pitt.edu]  United States             US   
3289               [upj.pitt.edu]  United States             US   
3290               [upb.pitt.edu]  United States             US   
3404                  [wcupa.edu]  United States             US   
3457                    [ycp.edu]  United States             US   
4189          [pennhighlands.edu]  United States             US   
4190                    [pit.edu]  United States             US   
4191                    [pti.edu]  United States             US   
4194               [upt.pitt.edu]  United States             US   
4399              [pointpark.edu]  United States             US   

                           web_pages state-province  \
14           [https://www.upmc.com/]   Pennsylvania   
197         [http://www.bloomu.edu/]   Pennsylvania   
243            [http://www.cup.edu/]   Pe

The second and third columns of my dataframe:

In [9]:
df.iloc[:, [1,2]]

country alpha_two_code
0     United States             US
1     United States             US
2     United States             US
3     United States             US
4     United States             US
...             ...            ...
4419  United States             US
4420  United States             US
4421  United States             US
4422  United States             US
4423  United States             US

[4424 rows x 2 columns]

First 4 rows of my dataframe:

In [10]:
df.iloc[0:4]

domains        country alpha_two_code  \
0    [marywood.edu]  United States             US   
1  [lindenwood.edu]  United States             US   
2    [sullivan.edu]  United States             US   
3        [fscj.edu]  United States             US   

                      web_pages state-province  \
0     [http://www.marywood.edu]           None   
1  [http://www.lindenwood.edu/]           None   
2       [https://sullivan.edu/]           None   
3       [https://www.fscj.edu/]           None   

                                    name  
0                    Marywood University  
1                  Lindenwood University  
2                    Sullivan University  
3  Florida State College at Jacksonville